torch/lib/c10d/CMakeLists.txt

```diff
@@ -1,8 +1,52 @@
 cmake_minimum_required(VERSION 3.2 FATAL_ERROR)
-set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake ${CMAKE_MODULE_PATH})
+
+# Find modules.
+# Note: this does NOT include <root>/cmake/Modules, because that woud
+# make find_package resolve to <root>cmake/Modules/FindGloo.cmake
+# instead of resolving to ../tmp_install/share/cmake/Gloo.
+list(APPEND CMAKE_MODULE_PATH
+  /usr/lib/x86_64-linux-gnu/
+  ${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake/public
+  ${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake/Modules_CUDA_fix)
+
+# Polyfill for upstream FindCUDA
+include(CMakeInitializeConfigs)
+
+# Relies on CMAKE_INSTALL_PREFIX to be set to ../tmp_install.
+# It then finds $PREFIX/share/cmake/ATen/ATenConfig.cmake,
+# which defines ATEN_INCLUDE_DIR and ATEN_LIBRARIES.
+find_package(ATen REQUIRED)
+if(NOT ATen_FOUND)
+  message(FATAL_ERROR "ATen not found")
+endif()
+
+find_package(Gloo REQUIRED)
+if(NOT Gloo_FOUND)
+  message(FATAL_ERROR "Gloo not found")
+endif()
+
+find_package(CUDA REQUIRED)
```

Comment:
CUDA and gloo are required only for now, and we'll make them optional in the future, right?

Reply:
Yep, that's the plan. Don't yet know how, but I want to have backend specific libraries like ATen.